bundles/core/src/main/java/com/adobe/cq/wcm/core/components/util/ComponentUtils.java

```diff
@@ -53,24 +54,50 @@ public final class ComponentUtils {
      */
     private static final int ID_HASH_LENGTH = 10;
 
+    
+    private static final String DATALAYER_ATTRIBUTE_NAME = "isDataLayerEnabled";
+    
     /**
      * Private constructor to prevent instantiation of utility class.
      */
     private ComponentUtils() {
         // NOOP
     }
 
+    /**
+     * Check if data layer is enabled.
+     *
+     * @param request  The request in which context this happens
+     * @param resource The resource to check.
+     * @return True if data layer is enabled for this resource, false otherwise.
+     */
+    public static boolean isDataLayerEnabled(@Nullable SlingHttpServletRequest request, 
```

Comment:
I think this can lead to some misbehaviour, as the CA configuration is resolved based on the resource path and the value is stored in the current request. It would be possible to include a resource into the current request which is located into a path were the CA configuration is different as for the path the current request is made against.
Example:
Request is against `/content/page1` where data layer is enabled. page1 includes a resource from `/content/page2` where data layer is disabled. In this case the resource from page2 would not have the data layer enabled. This might be fine but can also lead to some unexpected behaviour.
I wonder if there is no caching layer in the CA config resolver. If not, it would be a nice enhancement IMHO.

Reply:
Yes, I think the datalayer on/off switch should be based on the context of the current page. But the page can have a child resource which makes a include on a different context setting. At this is IMHO a problem which can't be solved by just checking for a request attribute.

Reply:
Sure. I am just wondering if a per-component setting would make sense. A requested page has a single datalayer object, which is dependent on the page (and to some extent on its content) itself. In my opinion it does not really make sense to allow every included object to come with its own datalayer "policy".

Because it would be extremely hard to debug for an author, why a certain piece of the page, which should be reflected in the datalayer, is not reflected but present in the page. Or the other way around. Because it would force the author to check all embedded elements.

I think that it's the easiest to explain and understand if we have a single per-page setting (can be inherited and overriden) to define if the datalayer is enabled or not. And then just follow that policy on that page, no matter where the data rendered on the page comes from or if the included object has a different policy defined.